cloudflare.go

```diff
@@ -35,6 +35,20 @@ func NewZone() *Zone {
 	return &Zone{}
 }
 
+// ZoneIDByName retrieves a zone's ID from the name.
+func (api *API) ZoneIDByName(zoneName string) (string, error) {
+	res, err := api.ListZones(zoneName)
+	if err != nil {
+		return "", err
+	}
+	for _, zone := range res {
+		if zone.Name == zoneName {
```

Comment:
Hmm, I'm not sure that's something I have an easy way of testing. Mostly I tried to improve upon the previous check of just taking the first zone returned by `ListZones`, but we can definitely add more checks as needed.